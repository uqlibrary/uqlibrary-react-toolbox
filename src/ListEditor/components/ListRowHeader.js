import React, {Component} from 'react';
import PropTypes from 'prop-types';
import FontIcon from 'material-ui/FontIcon';
import IconButton from 'material-ui/IconButton';
import {ConfirmDialogBox} from '../../index';

export default class ListRowHeader extends Component {
    static propTypes = {
        onDeleteAll: PropTypes.func.isRequired,
        locale: PropTypes.object,
        disabled: PropTypes.bool,
        hideReorder: PropTypes.bool
    };

    static defaultProps = {
        locale: {
            nameColumn: 'Name',
            reorderColumn: 'Reorder items',
            deleteAll: 'Remove all items',
            deleteAllConfirmation: {
                confirmationTitle: 'Delete all',
                confirmationMessage: 'Are you sure you want to delete all items?',
                cancelButtonLabel: 'No',
                confirmButtonLabel: 'Yes'
            }
        }
    };

    constructor(props) {
        super(props);
    }

    showConfirmation = () => {
        this.confirmationBox.showConfirmation();
    }

    render() {
        const {nameColumn, reorderColumn, deleteAll, deleteAllConfirmation} = this.props.locale;

        return (
            <div className="columns is-gapless is-mobile listHeader datalist datalist-header">
                <ConfirmDialogBox
                    onRef={ref => (this.confirmationBox = ref)}
                    onAction={this.props.onDeleteAll}
                    locale={deleteAllConfirmation}/>
                <div className="column name datalist-title">{nameColumn}</div>
                {
                    !this.props.hideReorder &&
                    <div className="column is-narrow is-hidden-mobile order datalist-title">{reorderColumn}</div>
                }
                <div className="column is-narrow buttons datalist-buttons">
                    <IconButton
                        tooltip={deleteAll}
                        onTouchTap={this.showConfirmation}
                        disabled={this.props.disabled}>
                        <FontIcon className="material-icons">delete_forever</FontIcon>
                    </IconButton>
                </div>
            </div>
        );
    }
}
