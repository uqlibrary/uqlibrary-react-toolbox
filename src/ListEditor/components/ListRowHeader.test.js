jest.dontMock('./ListRowHeader');

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ListRowHeader from './ListRowHeader';

function setup({onDeleteAll, disabled}) {

    const props = {
        onDeleteAll: onDeleteAll || jest.fn(), // : PropTypes.func.isRequired,
        // locale, // : PropTypes.object,
        disabled: disabled || false // : PropTypes.bool
    };

    return shallow(<ListRowHeader {...props} />);
}

describe('ListRowHeader renders ', () => {

    it('header for contributor editor control with name and delete all button only', () => {
        const wrapper = setup({ });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('header for contributor editor control with delete all disabled', () => {
        const wrapper = setup({ disabled: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
