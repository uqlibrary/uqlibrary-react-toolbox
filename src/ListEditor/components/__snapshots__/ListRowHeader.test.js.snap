// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRowHeader renders  header for contributor editor control with delete all disabled 1`] = `
<div
  className="columns is-gapless is-mobile listHeader datalist datalist-header"
>
  <ConfirmDialogBox
    hideCancelButton={false}
    locale={
      Object {
        "cancelButtonLabel": "No",
        "confirmButtonLabel": "Yes",
        "confirmationMessage": "Are you sure you want to delete all items?",
        "confirmationTitle": "Delete all",
      }
    }
    onAction={[MockFunction]}
    onRef={[Function]}
  />
  <div
    className="column name datalist-title"
  >
    Name
  </div>
  <div
    className="column is-narrow is-hidden-mobile order datalist-title"
  >
    Reorder items
  </div>
  <div
    className="column is-narrow buttons datalist-buttons"
  >
    <IconButton
      disableTouchRipple={false}
      disabled={true}
      iconStyle={Object {}}
      onTouchTap={[Function]}
      tooltip="Remove all items"
      tooltipPosition="bottom-center"
      touch={false}
    >
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        delete_forever
      </FontIcon>
    </IconButton>
  </div>
</div>
`;

exports[`ListRowHeader renders  header for contributor editor control with name and delete all button only 1`] = `
<div
  className="columns is-gapless is-mobile listHeader datalist datalist-header"
>
  <ConfirmDialogBox
    hideCancelButton={false}
    locale={
      Object {
        "cancelButtonLabel": "No",
        "confirmButtonLabel": "Yes",
        "confirmationMessage": "Are you sure you want to delete all items?",
        "confirmationTitle": "Delete all",
      }
    }
    onAction={[MockFunction]}
    onRef={[Function]}
  />
  <div
    className="column name datalist-title"
  >
    Name
  </div>
  <div
    className="column is-narrow is-hidden-mobile order datalist-title"
  >
    Reorder items
  </div>
  <div
    className="column is-narrow buttons datalist-buttons"
  >
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      onTouchTap={[Function]}
      tooltip="Remove all items"
      tooltipPosition="bottom-center"
      touch={false}
    >
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        delete_forever
      </FontIcon>
    </IconButton>
  </div>
</div>
`;
